examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+      "    assoc_mem = spa.AssociativeMemory(input_vocab=vocab)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "After the network has been created, we can test its functionality by presenting a semantic pointer at the input and comparing it with a semantic pointer at the output after running a simulation. The comparison is done by calculating the dot product. If the similarity between the output vector and the input vector is close to one, we can say that the associative memory successfully retrieved the pattern.\n",
+      "To achieve this task, we need to introduce two additional objects: `nengo.Node` and `nengo.Probe`. The `nengo.Node` object provides input to the associative memory. \n",
+      "`nengo.Probe` is used to gather simulation data, in our case the input being fed to the network (input pattern) and the network output (output pattern) at every simulation time step.\n",
+      "In this example, the input to the network is the semantic pointer `APPLE`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "with model1:\n",
+      "    input_vector = vocab['APPLE'].v\n",
+      "    input_node = nengo.Node(output=input_vector, label='input')\n",
+      "    \n",
+      "    nengo.Connection(input_node, assoc_mem.input)\n",
+      "\n",
+      "    input_probe = nengo.Probe(input_node)\n",
+      "    output_probe = nengo.Probe(assoc_mem.output, synapse=0.03)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We run the simulation for the 0.2 simulation time:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "sim = nengo.Simulator(model1)\n",
+      "sim.run(0.2)\n",
+      "t = sim.trange()\n",
+      "print('Number of simulation steps: %d' % len(t))"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We use the probes to get the data produced by the simulation:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "input_net = sim.data[input_probe]\n",
+      "output_net = sim.data[output_probe]\n",
+      "print(\"Input dimensions: (%d, %d)\" % input_net.shape)\n",
+      "print(\"Output dimensions: (%d, %d)\" % output_net.shape)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We see that for each simulation step (rows of the `input_net` and the `output_net` arrays) there is a 64-dimensional pattern represented in the associative memory. \n",
```

Comment:
32-dimensional
Also it is explicitely set before to 32, so I am not sure if it is necessary to reiterate this fact here and if so, it might be worth explaining that the other array dimension is the number of time steps.

Furthermore, I might name `*_net` `*_data` because it could get confused with a network.